citrix_hypervisor/tests/test_citrix_hypervisor.py

```diff
@@ -0,0 +1,81 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import logging
+
+import mock
+import pytest
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.citrix_hypervisor import CitrixHypervisorCheck
+from datadog_checks.dev.utils import get_metadata_metrics
+
+from .common import SESSION_MASTER, _assert_standalone_metrics, mocked_xenserver
+
+pytestmark = pytest.mark.unit
+
+
+@pytest.mark.parametrize(
+    'side_effect, expected_session, tag',
+    [
+        pytest.param([mocked_xenserver('error')], {}, []),
+        pytest.param([mocked_xenserver('master')], SESSION_MASTER, ['server_type:master']),
+        pytest.param([mocked_xenserver('slave'), mocked_xenserver('master')], SESSION_MASTER, ['server_type:slave']),
+        pytest.param([mocked_xenserver('slave'), mocked_xenserver('error')], {}, ['server_type:slave']),
+        pytest.param(mock.Mock(side_effect=Exception('Error')), {}, []),
+        pytest.param([mocked_xenserver('slave'), mock.Mock(side_effect=Exception('Error'))], {}, ['server_type:slave']),
+    ],
+)
+def test_open_session(instance, side_effect, expected_session, tag):
+    with mock.patch('six.moves.xmlrpc_client.Server', side_effect=side_effect):
+        check = CitrixHypervisorCheck('citrix_hypervisor', {}, [instance])
+        session = check.open_session()
+
+        assert session == expected_session
+        assert tag == check._additional_tags
+
+
+@pytest.mark.usefixtures('mock_responses')
+@pytest.mark.parametrize('server_type', [pytest.param('master'), pytest.param('slave')])
+def test_check(aggregator, instance, server_type):
+    with mock.patch('six.moves.xmlrpc_client.Server', return_value=mocked_xenserver(server_type)):
+        check = CitrixHypervisorCheck('citrix_hypervisor', {}, [instance])
+        check.check(instance)
```

Comment:
Let's try to only use this: https://datadoghq.dev/integrations-core/ddev/plugins/#check-execution